lib/jmap/mail/email/query/query_email_method.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:jmap_dart_client/http/converter/account_id_converter.dart';
+import 'package:jmap_dart_client/http/converter/id_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/unsigned_int_nullable_converter.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/capability/capability_identifier.dart';
+import 'package:jmap_dart_client/jmap/core/method/request/query_method.dart';
+import 'package:jmap_dart_client/jmap/core/request/request_invocation.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_comparator.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_filter_condition.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+class QueryEmailMethod extends QueryMethod with OptionalCollapseThreads {
+
+  QueryEmailMethod(AccountId accountId) : super(accountId);
+
+  @override
+  MethodName get methodName => MethodName('Email/query');
+
+  @override
+  Set<CapabilityIdentifier> get requiredCapabilities => {
+    CapabilityIdentifier.jmapCore,
+    CapabilityIdentifier.jmapMail
+  };
+
+  @override
+  List<Object?> get props => [methodName, accountId, limit, filter, sort];
+
+  factory QueryEmailMethod.fromJson(Map<String, dynamic> json) {
```

Comment:
IMO, we don't need to write the `fromJson` to `Method` :)

Reply:
Really? It contains an object that cannot be converted using JsonSerializable